### PR TITLE
Data model update

### PR DESCRIPTION
**Pull request by jimmymathews**
Addresses #94 . See the discussion of point-by-point progress there.

### CODE REVIEW

**Comment by CarlinLiao** [comment]
`dataset1/diagnosis.tsv` has `1` in the "Date of diagnosis" column, but `dataset2/diagnosis.tsv` has `timepoint 2` in the same column. Is the variance in formatting intentional?

**Comment by jimmymathews** [comment]
Hm I don't see what you're referencing. In both cases I am seeing blank "Date of diagnosis".
The "Last date of considered evidence" has "timepoint 3" and "timepoint 2" respectively.

For dataset 1, the sequence of events is:
1. Collect some specimens
2. Do injections
3. Collect more specimens
4. Do diagnoses

For dataset 2, the sequence of events is:
1. Collect some specimens
2. Do treatment
3. Do diagnoses

**Comment by CarlinLiao** [comment]
My bad, I can't read. These do give me the right timepoints to finish porting the melanoma and breast cancer datasets.

File: spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset2/diagnosis.tsv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Subject of diagnosis	Diagnosed condition	Diagnosis	Date of diagnosis	Last date of considered evidence
+Jackson2020_1	Response to hormone therapy	Refractory		timepoint 2
+Jackson2020_3	Response to hormone therapy	Refractory		timepoint 2
+Jackson2020_4	Response to hormone therapy	Sensitive		timepoint 2
+Jackson2020_5	Response to hormone therapy	Sensitive		timepoint 2
```
[the remark this answers is not in the document]
Comment (jimmymathews) [reply spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset2/diagnosis.tsv:5]: Yes, this is deliberate. Remember it is the "Date of evidence" considered by the diagnosis, not the diagnosis act. In this dataset, I think the last evidence considered (as far as we know) is at treatment time, there being no further specimens collected. Come to think of it I think it is worth checking the manuscript to see that the samples really are pre-treatment.
Similarly in the melanoma case I'm using "timepoint 3" for both "collecting more specimens" date and "date of last evidence for diagnosis".

File: spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset1/samples.tsv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Sample ID	Source subject	Source site	Source subject age at specimen collection	Extraction method	Extraction date	Preservation method	Storage location	Assay	Assessment
+lesion 0_1	0	Left groin	56 years	Surgical excision	timepoint 1	Formalin-fixed and paraffin-embedded	MSKCC Department of Pathology temperature controlled storage units		
+lesion 0_2	0	Left foot	56 years	Surgical excision	21 days post-injection, timepoint 3	Formalin-fixed and paraffin-embedded	MSKCC Department of Pathology temperature controlled storage units	Intralesional IL-2 response	Non-complete response
+lesion 0_3	0	Left groin	56 years	Surgical excision	28 days post-injection, timepoint 3	Formalin-fixed and paraffin-embedded	MSKCC Department of Pathology temperature controlled storage units	Intralesional IL-2 response	Complete response
```
[the remark this answers is not in the document]
Comment (jimmymathews) [reply spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset1/samples.tsv:4]: The timepoint signifiers are meant to be subject-specific.

File: spatialprofilingtoolbox/workflow/source_file_adi_parsing/samples.py [hunk 3]
```diff
@@ -45,6 +42,8 @@ def parse(self, connection, fields, samples_file, age_at_specimen_collection, fi
             )
 
         for i, sample in samples.iterrows():
```
[the remark this answers is not in the document]
Comment (jimmymathews) [reply spatialprofilingtoolbox/workflow/source_file_adi_parsing/samples.py:44]: I suppose they could be combined, but they do different things. The `sample` input row has lots of information put together, and these two loops extract independent parts.
Making it one loop reduces the number of loops (good) but also creates coupling between the "make call specimen collection process records" and "make all histology assessment process records" functions (bad).

File: spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset1/samples.tsv (hunk 2 above)
[the remark this answers is not in the document]
Comment (jimmymathews) [reply spatialprofilingtoolbox/test_data/adi_preprocessed_tables/dataset1/samples.tsv:4]: ... although these are the same subject. So yes, this is an error.

**Comment by jimmymathews** [comment]
After discussion, we found the convenience `sample_strata` as described above to be somewhat inconsistent. Including the "timepoint dd" information in stratum descriptors implicitly claims common relevance of that timepoint to each possible sample falling under the description, which was not meant to be the case. Indeed, there seems to be no general-purpose matching possible between the chronologies across patients, and we probably should not attempt to impose one.
What is chronologically general-purpose, however, and what will meet our current needs, is a lower-resolution variant: just recording whether a sample was extracted (i) before all interventions, (ii) after all interventions, (iii) between some interventions, and (iv) from a subject for whom no intervention was made.
I am updating this accordingly.

**Comment by CarlinLiao** [comment]
I think that's it. Reviewed and approved.

**Comment by jimmymathews** [comment]
Hurray! And all tests pass.

**Pull request merged**